modules/shortcodes/email-subscribe.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ **/
+
+/**
+ * Class Jetpack_Email_Subscribe
+ * This class encapsulates shortcode for subscribing to a MailChimp list.
+ * It displays a simple signup form that gets an email address from the user and signs him for a list
+ * selected in "Sharing" section in calypso.
+ * Other Email services can be implemented as well in the future.
+ */
+class Jetpack_Email_Subscribe {
+
+	private static $shortcode = 'jetpack-email-subscribe';
+
+	private static $css_classname_prefix = 'jetpack-email-subscribe';
+
+	private static $instance;
+
+	private static $version = '1.0';
+
+	private function __construct() {
+	}
+
+	/**
+	 * This follows a classic singleton pattern.
+	 *
+	 * @return Jetpack_Email_Subscribe|null
+	 */
+	public static function get_instance() {
+		// Do not load this at all if it's neither a WPCOM or a connected JP site.
+		if ( ! ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || Jetpack::is_active() ) ) {
```

Comment:
I would check if the Subscriptions module is added as well, since you rely on it in the shortcode.

Reply:
I am not relying on it. It *does not* require subscriptions module to be active to work :)

Reply:
Ok then. Do you think you could update the testing instructions so we can now what to expect when signing up with that shortcode: where is your email address sent when subscribing there, what emails should be sent to you after you've subscribed?

Reply:
Updated!